src/interface/variable_pack.hpp

```diff
@@ -95,9 +95,9 @@ class VariablePack {
     return sparse_ids_(n);
   }
   KOKKOS_FORCEINLINE_FUNCTION
-  int GetGlobalSparseId(const int n) const { return GetSparse(n); }
+  int GetSparseId(const int n) const { return GetSparse(n); }
   KOKKOS_FORCEINLINE_FUNCTION
-  int GetLocalSparseId(const int n) const { return GetSparse(n); }
+  int GetSparseIndex(const int n) const { return GetSparse(n); }
```

Comment:
Shouldn't this just be `return n;`? I might be missing something.

Reply:
`n` is the offset of a variable in the pack and the sparse index that identifies which sparse item that variable belongs with can be different.  

@jdolence: can you confirm that my understanding is correct as I'm just coming into this.

Reply:
Not @jdolence but I think thats right

Reply:
yes that's right.  it is definitely not `n` in general.